rascal-vscode-extension/src/fs/VSCodeURIResolver.ts

```diff
@@ -368,6 +383,41 @@ class ResolverClient implements VSCodeResolverServer, Disposable  {
             };
         }
     }
+
+    async fileStat(req: ISourceLocationRequest): Promise<FileAttributesResult> {
+        const fileInfo = await this.stat(req);
+        return {
+            errorCode: 0,
+            exists: (await this.exists(req)).result!,
+            type: fileInfo.type.valueOf(),
+            ctime: fileInfo.ctime,
+            mtime: fileInfo.mtime,
+            size: fileInfo.size,
+            permissions: fileInfo.permissions ? fileInfo.permissions.valueOf() : 0,
+            isWritable: (await this.isWritable(req)).result!
```

Comment:
shouldn't  `isReadable` also be implemented?

Reply:
Good catch